File: python/1897.Redistribute-Characters-to-Make-All-Strings-Equal.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        freq = defaultdict(int)
        length = len(words)

        for w in words:
            for c in w:
                freq[c] += 1

        for val in freq.values():
            if val % length:
                return False

        return True

File: javascript/1897.Redistribute-Characters-to-Make-All-Strings-Equal.js
/**
 * @param {string[]} words
 * @return {boolean}
 */
var makeEqual = function (words) {
  const map = new Map();
  const length = words.length;

  const addToMap = (word) => {
    for (const c of word) {
      map.set(c, (map.get(c) || 0) + 1);
    }
  };

  for (const word of words) {
    addToMap(word);
  }

  for (const val of map.values()) {
    if (val % length) {
      return false;
    }
  }

  return true;
};
